### PR TITLE
Override knexfile options with CLI options

### PR DESCRIPTION
**Pull request by KoryNunn**
This PR allows overriding options in a passed knexfile with other options passed to the CLI.

### CODE REVIEW

**Comment by kibertoad** [comment]
There's a test failing, could you please take a look?

**Comment by kibertoad** [comment]
@KoryNunn Could you please also update documentation in https://github.com/knex/documentation?

File: bin/utils/cli-config-utils.js [hunk 1]
```diff
@@ -6,6 +6,28 @@ const tildify = require('tildify');
...
+
+  if (opts.client) {
+    config.client = opts.client;
+  }
+
+  if (opts.connection) {
+    config.connection = opts.connection;
+  }
+
+  if (opts.migrationsDirectory) {
```
Comment (kibertoad) [inline bin/utils/cli-config-utils.js:20]: I wonder if there is a way to make all overrides case-insensitive. Probably we can do check on lowercased entry pairs.

**Comment by kibertoad** [comment]
@KoryNunn Do you need any help with this?

File: bin/utils/cli-config-utils.js (hunk 1 above)
Comment (OlivierCavadenti) [reply bin/utils/cli-config-utils.js:21]: As I can understand, the goal is to lower case the properties in options object before merge with config in knexfile ?

**Comment by KoryNunn** [comment]
Hi just a quick update. I no longer use knex, and as such have no capability for valuable input. I'll leave this PR open in case it is of use to anyone else, but also not against it being closed.

File: bin/utils/cli-config-utils.js (hunk 1 above)
Comment (kibertoad) [reply bin/utils/cli-config-utils.js:21]: But some options are camelCase, no?

**Comment by OlivierCavadenti** [comment]
> Hi just a quick update. I no longer use knex, and as such have no capability for valuable input. I'll leave this PR open in case it is of use to anyone else, but also not against it being closed.

It's ok no problem at all, I will finish it !

File: bin/utils/cli-config-utils.js (hunk 1 above)
Comment (OlivierCavadenti) [reply bin/utils/cli-config-utils.js:21]: Mmh, I will check that this week.

**Comment by OlivierCavadenti** [comment]
@kibertoad 
Doc here https://github.com/knex/documentation/pull/395
By the way, I assume I can also close https://github.com/knex/knex/pull/5006 which I have opened but it's a duplicate of this one ?
For your comment with camelCase, I made some tests, but since cli options with commander have fixed case and transformed in camelCase properties, I don't see any problems, and unit test is ok. Maybe I miss something ?

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Yeah, looks good, and we can close 5006 too.

**Comment by kibertoad** [comment]
Released in 1.0.5

**Pull request merged**